src/client/components/ProductsView/ProductsView.component.js

```diff
@@ -16,11 +17,14 @@ export default function ProductsView({ products }) {
           {products.map((product) => (
             <li key={product.id} className="product-item">
               <Product
```

Comment:
if(product.is_on_on_discount===1){
<OfferProductModal 
 id={product.id}
                image={product.picture}
                name={product.name}
                price={product.price}
                onClick={addToCart}
                addToFavorites={addFavorites}
                discount={product.discount_percent}
                currency={product.currency}
              />
} else {
<Product
id={product.id}
                image={product.picture}
                name={product.name}
                price={product.price}
                onClick={addToCart}
                addToFavorites={addFavorites}
                currency={product.currency}
              />
}

Reply:
Hi Divya, I think it is not related to cart. It should ideally go into a separate issue and a separate PR. I tried implementing it in this PR using the above suggestion but then realized that OfferProduct is almost a copy of Product component. It might then be better to have the discount/Offer functionality built within product component itself which will then remove duplicity in few places. But better to do in a separate issue.